src/components/sbb-form-field/sbb-form-field.tsx

```diff
@@ -0,0 +1,180 @@
+import { AgnosticMutationObserver as MutationObserver } from '../../global/helpers/mutation-observer';
+
+let nextId = 0;
+
+/**
+ * @slot label - Slot to render a label.
+ * @slot prefix - Slot to render an icon on the left side of the input.
+ * @slot input - Slot to render an input.
+ * @slot suffix - Slot to render an icon on ther right side of the input.
+ * @slot error - Slot to render an error.
+ */
+@Component({
+  shadow: true,
+  styleUrl: './sbb-form-field.scss',
+  tag: 'sbb-form-field',
+})
+export class SbbFormField {
+  /**
+   * Add a specific space if the `<sbb-error>` is present.
+   */
+  @Prop() public errorSpace?: InterfaceSbbFormFieldAttributes['errorSpace'] = 'default';
+
+  /**
+   * Add a `<label>` for the input.
+   */
+  @Prop() public label: string;
+
+  /**
+   * Indicates whether the input is optional.
+   */
+  @Prop() public optional?: boolean;
+
+  /**
+   * Size variant, either l or m.
+   */
+  @Prop() public size?: InterfaceSbbFormFieldAttributes['size'] = 'l';
+
+  /**
+   * @internal
+   * It is used internally to get the native `readonly` attribute from `<input>`.
+   */
+  @State() private _readonly: boolean;
+
+  /**
+   * @internal
+   * It is used internally to get the native `disabled` attribute from `<input>`.
+   */
+  @State() private _disabled: boolean;
+
+  /**
+   * The internal `<sbb-form-field>` element.
+   */
+  @Element() private _element: HTMLElement;
+
+  /**
+   * @internal
+   * It is used internally to set the `for` attribute of the `<label>`.
+   */
+  private _id: string;
+
+  /**
+   * @internal
+   * It is used internally to set the `aria-describedby` attribute into the `input` whether the `<sbb-form-error>` is present.
+   */
+  private _idError: string;
+
+  /**
+   * @internal
+   * It is used internally to get the `input` slot.
+   */
+  private _input: HTMLInputElement;
+
+  /**
+   * @internal
+   * Listen the changes on `readonly` and `disabled` attributes of `<input>`.
+   */
+  private _formFieldAttributeObserver = new MutationObserver(this._onAttributesChange.bind(this));
+
+  public componentWillLoad(): void {
+    this._idError = this._element?.querySelector('[slot="error"]')?.getAttribute('id');
+    if (this._idError) {
+      this._element.querySelector('[slot="input"]').setAttribute('aria-describedby', this._idError);
+    }
+
+    const input = this._element.querySelector('[slot="input"]');
+    this._readonly = input.hasAttribute('readonly');
+    this._disabled = input.hasAttribute('disabled');
+
+    this._formFieldAttributeObserver.observe(input, {
+      attributes: true,
+      attributeFilter: ['readonly', 'disabled'],
+    });
+  }
+
+  public disconnectedCallback(): void {
+    this._formFieldAttributeObserver.disconnect();
+  }
+
+  /**
+   * @private
+   * It is used internally to assign the attributes of `<input>` to `_id` and `_input`.
+   */
+  private _onSlotInputChange(): void {
```

Comment:
Minor performance nitpick: The `slotchange` event provides an event object, which provides the attached content:
`(event.target as HTMLSlotElement).assignedElements()[0]`
This could be used instead of the querySelector

Reply:
Additional performance nitpick; Assign the input element to a variable instead of retrieving it each time.